features/smoke_tests/handled.feature

```diff
@@ -0,0 +1,164 @@
+    And the event "app.version" equals "1.1.14"
+    And the event "app.versionCode" equals 34
+    And the event "app.duration" is not null
+    And the event "app.durationInForeground" is not null
+    And the event "app.inForeground" is true
+    And the payload field "events.0.metaData.app.memoryUsage" is greater than 0
+    And the event "metaData.app.name" equals "MazeRunner"
+    And the event "metaData.app.lowMemory" is not null
+
+    # Device data
+    And the payload field "events.0.device.cpuAbi" is a non-empty array
+    And the event "device.jailbroken" is false
+    And the event "device.id" is not null
+    And the payload field "events.0.device.id" is stored as the value "device_id"
+    And the event "device.locale" is not null
+    And the event "device.manufacturer" is not null
+    And the event "device.model" is not null
+    And the event "device.osName" equals "android"
+    And the event "device.osVersion" is not null
+    And the event "device.runtimeVersions" is not null
+    And the event "device.runtimeVersions.androidApiLevel" is not null
+    And the event "device.runtimeVersions.osBuild" is not null
+    And the payload field "events.0.device.totalMemory" is greater than 0
+    And the payload field "events.0.device.freeDisk" is greater than 0
+    And the payload field "events.0.device.freeMemory" is greater than 0
+    And the event "device.orientation" equals "portrait"
+    And the event "device.time" is a timestamp
+    And the event "metaData.device.locationStatus" is not null
+    And the event "metaData.device.emulator" is false
+    And the event "metaData.device.networkAccess" is not null
+    And the event "metaData.device.charging" is not null
+    And the event "metaData.device.screenDensity" is not null
+    And the event "metaData.device.dpi" is not null
+    And the event "metaData.device.screenResolution" is not null
+    And the event "metaData.device.brand" is not null
+    And the event "metaData.device.batteryLevel" is not null
+
+    # User
+    And the event "user.id" is not null
+    And the payload field "events.0.user.id" equals the stored value "device_id"
+
+    # Breadcrumbs
+    And the event has a "state" breadcrumb named "Bugsnag loaded"
+    And the event has a "manual" breadcrumb named "HandledJavaSmokeScenario"
+    And the event "breadcrumbs.2.metaData.source" equals "BreadcrumbCallback"
+
+    # MetaData
+    And the event "metaData.TestData.ClientMetadata" is true
+    And the event "metaData.TestData.CallbackMetadata" is true
+
+Scenario: Notify Kotlin exception with overwritten configuration
+    When I run "HandledKotlinSmokeScenario"
+    Then I wait to receive a request
+    And the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+
+    # Exception details
+    And the payload field "events" is an array with 1 elements
+    And the exception "errorClass" equals "java.lang.RuntimeException"
```

Comment:
Could the scenario be changed to something like `IllegalStateException`? This would add extra assurance that Bugsnag isn't wrapping the captured exception with its own `Throwable` and that the original type is serialized correctly.

Reply:
What was the outcome here?

Reply:
This was addressed as suggested